### ISSUE TITLE
Unable to start Docker container

### ISSUE THREAD

**Issue by thomazio** [rogerfar/rdt-client#47]
OS: Linux
rdtclient version:  1.7.2

```
docker: Error response from daemon: OCI runtime create failed: container_linux.go:367: starting container process caused: exec: "dotnet": executable file not found in $PATH: unknown.
ERRO[0007] error waiting for container: context canceled 
```

**Comment by rogerfar** [comment]
I pulled a fresh copy on my Windows machine and it worked fine, what host OS are you using and how do you start the container?

**Comment by timcoop3r** [comment]
fwiw, I've also pulled 1.7.2 and have no issues running on my Fedora box.

**Comment by thomazio** [comment]
> what host OS are you using and how do you start the container?

Arch Linux ARM

Start
```# docker start rdtclient```

Or to spin up a new container
```# docker run --pull=always --volume /mnt/hdd1/Downloads/:/data/downloads --volume /mnt/hdd1/Downloads/:/data/db --log-driver json-file --log-opt max-size=10m -p 6500:6500 --name rdtclient rogerfar/rdtclient:latest

**Comment by Reen68** [comment]
Started to have this behaviour, too as of today.

I have an open issue myself for another issue and just wanted to pull the rdtclient again and it gives me the same error - everything was working yesterday (installed it 2 times yesterday, same way, same docker-compose file etc.)

**Comment by rogerfar** [comment]
What is your docker host?

**Comment by Reen68** [comment]
Using a Raspbi with Debian.

I was troubleshooting my own issue yesterday evening and therefore reinstalled 2-3 times via the same docker compose (only changed volumes around) as of today I get the same error message.

Something with your latest changes I guess?

**Comment by rogerfar** [comment]
Yeah I used a new docker build file, but it looks like it's not installing the dotnet tools for arm properly, if you're adventures you can login the container and install dotnet until I figure it out..

**Comment by ravensorb** [comment]
@Reen68, let me see if I can reproduce and fix on my raspbi.  I think I see what the issue is so just need to confirm

**Comment by Reen68** [comment]
> Yeah I used a new docker build file, but it looks like it's not installing the dotnet tools for arm properly, if you're adventures you can login the container and install dotnet until I figure it out..

I'm completely new to all of this and learn a lot by trial and error but i actually tried to pull the NET5 Runtime via

docker pull mcr.microsoft.com/dotnet/runtime

But even after that I had the same behaviour if thats of any help to you.

**Comment by ravensorb** [comment]
@Reen68  are cloning the git repo and then running ```docker build```?  or are you running this command?

```
docker run --pull=always
		   --volume /your/download/path/:/data/downloads \
		   --volume /your/storage/path/:/data/db \
		   --log-driver json-file \
		   --log-opt max-size=10m \
		   -p 6500:6500 \
		   --name rdtclient \
		   rogerfar/rdtclient:latest
```

**Comment by Reen68** [comment]
I'm actually using a docker-compose with

version: '3.3'
services:
    rdtclient:
        container_name: rdtclient
        volumes:
            - 'D:/Downloads/:/data/downloads'
            - 'D:/Docker/rdt-client/:/data/db'
        image: rogerfar/rdtclient
        restart: always
        logging:
            driver: json-file
            options:
                max-size: 10m
        ports:
            - '6500:6500'

**Comment by ravensorb** [comment]
Thanks -- that helps.  Let me look into it.

**Comment by ZuurtjeNL** [comment]
> Yeah I used a new docker build file, 

This is maybe the reason it's working for me now on a Synology NAS, before I had issues.

**Comment by rogerfar** [comment]
I have reverted 1.7.3 to the old build until we can figure out the ARM issue.

**Comment by ravensorb** [comment]
@rogerfar what version SDK are you building against?  I think I see what happened -- just want to confirm my testing (right now i am getting a core dump which made me wonder about the framework version)

**Comment by ravensorb** [comment]
@rogerfar i just finished a first pass at the fix and have tested it on both a standard x64 server as well as an rpi4.  I moved just the docker related stuff into their own repo to make it easier to work with (for both of us) and you can find it here -- https://github.com/ravensorb/docker-rdt-client 

This repo has a README that explains the setup, it also has the Dockerfile, a sample compose file, a ps1 and bat file to help with the build and a root folder that contains a key file to set up rdt-client as a service inside the container (note the new image does not use an entry point or command anymore).  

Take a look and let me know what you think.

**Comment by rogerfar** [comment]
I think that looks good, do you want to create a pull request so I can merge it in and try it out?

**Comment by ravensorb** [comment]
Just created the pr

**Comment by rogerfar** [comment]
This is a bit of a dupe of #51 but the new version is pushed and Arm7 should be all good now.

**Comment by farnir** [comment]
Hi, got also an issue since the dockerfile upgrade. The container start but then the application keep looping with that logs:
`
Changing to /app folder
Starting rtdclient
Changing to /app folder
Starting rtdclient
Changing to /app folder
Starting rtdclient
`
I'm using the lastest docker image rogerfar/rdtclient: 1.7.4
Running in docker 20.10.6 on dietpi 7.0.2 on rasp4.
There is how I start the container:
`docker run --restart unless-stopped --user root --volume /mnt/dietpi_userdata/downloads/:/data/downloads --volume /mnt/dietpi_userdata/real_debrid/:/data/db --log-driver json-file --log-opt max-size=10m -p 6500:6500 --detach --name rdtclient rogerfar/rdtclient:latest`

**Comment by ravensorb** [comment]
Try removing the "--user root" part of the run command

**Comment by farnir** [comment]
> 
> 
> Try removing the "--user root" part of the run command
I have the same issue even without root.

**Comment by ravensorb** [comment]
@farnir two thoughts -- can you clear the changed containers and images for rdtclient and the force a pull of the latest image?

```
docker rmi -f $(docker images --filter=reference="rogerfar/*:*" -q)
docker run --restart unless-stopped --volume /mnt/dietpi_userdata/downloads/:/data/downloads --volume /mnt/dietpi_userdata/real_debrid/:/data/db --log-driver json-file --log-opt max-size=10m -p 6500:6500 --detach --name rdtclient rogerfar/rdtclient:latest
```

**Comment by ravensorb** [comment]
Its ok if the uid/guid doesn't exist in that it will not cause any errors (just makes accessing them a little tricky outside the container).  That said, the defaults in the new image are 1000:1000 so not sure why you are seeing 911.  Are you overriding them in an .env file?

**Comment by rogerfar** [comment]
Does it create a DB and log file? Or it doesn't get that far?

**Comment by farnir** [comment]
No the directory is empty.

**Comment by rogerfar** [comment]
What UID is that downloads folder on when it gets created? Also 911? And does docker logs show anything?

**Comment by farnir** [comment]
Yes, both downloads and db are on 911:911. And the logs are quite empty because all I got is: (like x100)
```
Changing to /app folder
Starting rtdclient
Changing to /app folder
Starting rtdclient
```

**Comment by ravensorb** [comment]
@farnir  here is my info

```
PRETTY_NAME="Raspbian GNU/Linux 10 (buster)"
NAME="Raspbian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
```

```
Linux raspberrypi 5.10.17-v7l+ #1403 SMP Mon Feb 22 11:33:35 GMT 2021 armv7l GNU/Linux
```

```
Client: Docker Engine - Community
 Version:           20.10.6
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        370c289
 Built:             Fri Apr  9 22:46:18 2021
 OS/Arch:           linux/arm
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.6
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       8728dd2
  Built:            Fri Apr  9 22:44:17 2021
  OS/Arch:          linux/arm
  Experimental:     false
 containerd:
  Version:          1.4.4
  GitCommit:        05f951a3781f4f2c1911b05e61c160e9c30eaa8e
 runc:
  Version:          1.0.0-rc93
  GitCommit:        12644e614e25b05da6fd08a38ffa0cfe1903fdec
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

```
docker-compose version 1.29.1, build unknown
```

**Comment by ravensorb** [comment]
@farnir question -- can you bash into the container, change to the /app folder and run the following 

```
dotnet RdtClient.Web.dll
```

Do you get any error info when you do that?

**Comment by farnir** [comment]
@ravensorb  When doing that I got this error :
```
root@DietPi:~# docker exec -it rdtclient /bin/bash
root@e1af29e2e88e:/app# dotnet RdtClient.Web.dll
Aborted (core dumped)
root@e1af29e2e88e:/app# 
```

**Comment by ravensorb** [comment]
Hmm, it could be that the the image that was downloaded doesn't match your architecture (ex: trying to run an amd64 on an arm device).  If you do a docker inspect on your image, what do you see?

**Comment by farnir** [comment]
Here is the docker inspect on the image:
```
[
    {
        "Id": "sha256:89e9701c9e5017843ba16b182f7c9081a3690db440cf71649c8eef88700d7b18",
        "RepoTags": [
            "rogerfar/rdtclient:latest"
        ],
        "RepoDigests": [
            "rogerfar/rdtclient@sha256:a3f4462dcb838fc19c476e92d9f0baa80bfcd6f9e5133591bb6637c6bdd59b28"
        ],
        "Parent": "",
        "Comment": "buildkit.dockerfile.v0",
        "Created": "2021-04-28T14:02:19.6473872Z",
        "Container": "",
        "ContainerConfig": {
            "Hostname": "",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": null,
            "Cmd": null,
            "Image": "",
            "Volumes": null,
            "WorkingDir": "",
            "Entrypoint": null,
            "OnBuild": null,
            "Labels": null
        },
        "DockerVersion": "",
        "Author": "",
        "Config": {
            "Hostname": "",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "ExposedPorts": {
                "6500/tcp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/share/dotnet",
                "HOME=/root",
                "LANGUAGE=en_US.UTF-8",
                "LANG=en_US.UTF-8",
                "TERM=xterm",
                "TARGETPLATFORM=linux/arm/v7",
                "BUILDPLATFORM=linux/amd64",
                "XDG_CONFIG_HOME=/config/xdg",
                "RDTCLIENT_BRANCH=main"
            ],
            "Cmd": null,
            "Image": "",
            "Volumes": {
                "/config": {},
                "/data": {}
            },
            "WorkingDir": "/app",
            "Entrypoint": [
                "/init"
            ],
            "OnBuild": null,
            "Labels": {
                "build_version": "Linuxserver.io extended version:-  Build-date:- ",
                "maintainer": "ravensorb",
                "org.opencontainers.image.authors": "linuxserver.io",
                "org.opencontainers.image.created": "2021-04-27T06:06:18+01:00",
                "org.opencontainers.image.description": "baseimage-ubuntu image by linuxserver.io",
                "org.opencontainers.image.documentation": "https://docs.linuxserver.io/images/docker-baseimage-ubuntu",
                "org.opencontainers.image.licenses": "GPL-3.0-only",
                "org.opencontainers.image.ref.name": "d03eb84840403eee0cf526d13605df46a43e59bb",
                "org.opencontainers.image.revision": "d03eb84840403eee0cf526d13605df46a43e59bb",
                "org.opencontainers.image.source": "https://github.com/linuxserver/docker-baseimage-ubuntu",
                "org.opencontainers.image.title": "Baseimage-ubuntu",
                "org.opencontainers.image.url": "https://github.com/linuxserver/docker-baseimage-ubuntu/packages",
                "org.opencontainers.image.vendor": "linuxserver.io",
                "org.opencontainers.image.version": "f5630821-ls38"
            }
        },
        "Architecture": "arm",
        "Variant": "v7",
        "Os": "linux",
        "Size": 326490993,
        "VirtualSize": 326490993,
        "GraphDriver": {
            "Data": {
                "LowerDir": "/mnt/dietpi_userdata/docker-data/overlay2/c0ec46464718a7941672a6be7da5a90893ef7e0c02b50d5efa15da59338f82ea/diff:/mnt/dietpi_userdata/docker-data/overlay2/a62ad6bce16384c5113a66b8d07c3a87739dcd3364f0ba4d42ac690f4dcbe61a/diff:/mnt/dietpi_userdata/docker-data/overlay2/1e494aa19b566a01f85ffad1474065af3982840255ed42a94a29b00bd1d7f335/diff:/mnt/dietpi_userdata/docker-data/overlay2/a1f07532ca25ad86847033fdf203af2ac004d076c4fc33e60648dde43711617e/diff:/mnt/dietpi_userdata/docker-data/overlay2/116b19e5ee96247414936229699f1928b9069fb568c73a2782055cc0bbbdd5eb/diff:/mnt/dietpi_userdata/docker-data/overlay2/2d515581a13c7febc8eb4d73c98c09f8acc531ace0bf11fed77cdc89d59a371b/diff:/mnt/dietpi_userdata/docker-data/overlay2/ac121ad74828b00a353c998a9943ff6ef88d39099fadb70d077d3c837306fc86/diff:/mnt/dietpi_userdata/docker-data/overlay2/868f192a7cb25e9d42a56cb9340a6a1fa69ca56aa772dc5a5d2a9bd2b6076e93/diff:/mnt/dietpi_userdata/docker-data/overlay2/aabdcbf568d94a64ad5b59bd9861bd56cfe3a4e7b378981e74ccfc31fb0b7fa0/diff:/mnt/dietpi_userdata/docker-data/overlay2/d935aca3a7ba7f634807bec9cf4a5abb9d0564486f767f3a3a36119b29c55c22/diff",
                "MergedDir": "/mnt/dietpi_userdata/docker-data/overlay2/09f3bb44a60ab04e82b35001486b11d207643cfa4f8cff6258475527590b653e/merged",
                "UpperDir": "/mnt/dietpi_userdata/docker-data/overlay2/09f3bb44a60ab04e82b35001486b11d207643cfa4f8cff6258475527590b653e/diff",
                "WorkDir": "/mnt/dietpi_userdata/docker-data/overlay2/09f3bb44a60ab04e82b35001486b11d207643cfa4f8cff6258475527590b653e/work"
            },
            "Name": "overlay2"
        },
        "RootFS": {
            "Type": "layers",
            "Layers": [
                "sha256:3d8fe906856065c73d19e178c2bc7cf0f5bcbe25785be0c8e51273389fcc82ab",
                "sha256:aa8eab1e188bd5417cde6a9dc2bc5608d7034380c1a8faae49da51ba5225edd8",
                "sha256:633a7973d274fed07751c34675bcbf10d75d509e9ba5c03651b6ee5ef8dc4ff6",
                "sha256:c11b7c4347262bb1d09e01be2792739401ee3d52db04c76aa164f47bfc4d0372",
                "sha256:a424e78fa1abbd974d1ab4d517c6994f0f062aba8e43d4341952b294421b1970",
                "sha256:0def27620e07d94ef16276d9d892d51f1ccb864205d06f68b6797ec48e3df6b1",
                "sha256:8518cd24cbc799c48c70ba08ed5d3d6cd29222d9aec8779f0b13fafa4773ae41",
                "sha256:9f0b0b1ecd9d24d52a05ff9ce6c6299faad427a71a70e8cccc80b9234a111fbe",
                "sha256:adee7549f1704cdbddc3e88d63534a24b29f6423bb304c23f83019039731c18a",
                "sha256:8bc89784828d1d0e0cfde23b9bff7915256fefaf88b595675baeb48486b28a72",
                "sha256:32a9ff6af5c4167c0f5818e9b9de6650dd725a629468fad7edb6fdd252589708"
            ]
        },
        "Metadata": {
            "LastTagTime": "0001-01-01T00:00:00Z"
        }
    }
]
```

**Comment by timcoop3r** [comment]
fwiw, because the linuxserver.io base images are used now 911.911 is the default UID GID that their stuff sets.

If you want to change this you need to set it as such. Looks like from the other comments in the thread people may not have realised this?

That said, I don't specifically see this causing that problem, however @farnir you probably want to be setting it correctly regardless?

**Comment by ravensorb** [comment]
@farnir ok, this is a bit strange :)  The image looks correct.  Question, just for the heck of it -- can you try using a version i created?

```docker pull ravensorb/rdtclient```

**Comment by farnir** [comment]
Same this happen with the image **ravensorb/rdtclient**, I'm gone try with a clean install of my OS on the RPI4.

**Comment by quaresc** [comment]
I had the same issue and I remembered that my volume was under `root` permissions ! 
Simply start your container with `--privileged` option and you should be good 😉 !

**Comment by farnir** [comment]
I tried to run the container with `--privileged` and I worked with the latest version of rogerfar/rdtclient docker image !! 
Thanks @quaresc  !!

**Comment by thomazio** [comment]
Trying to run the latest Docker image and I get also this error:

```
Changing to /app folder
Starting rtdclient
Failed to load /app/libhostpolicy.so, error: /app/libhostpolicy.so: wrong ELF class: ELFCLASS32
An error occurred while loading required library libhostpolicy.so from [/app/]
```

I'm using Arch Linux ARM on arm64. Using the `--privileged` flag, it's still showing the above error message.

**Comment by rogerfar** [comment]
@ravensorb any idea why this could happen?

I think it might be because the actual build isn't triggered correctly for the ARM64 version? See https://github.com/dotnet/runtime/issues/3106

**Comment by rogerfar** [comment]
I spent a bit of time reworking the build scripts for Docker, I am now using a different build for the ARM64 version for dotnet.

Can  you try 1.7.6 and see if that works?

**Comment by ravensorb** [comment]
I'll also take a look as soon as I get near a computer later this week

**Comment by jschmaltz69** [comment]
Not sure if this is helpful at all, but I ran into this issue repeatedly on x64 Debian while trying to run an in-place upgrade from 1.7.2. I was able to get it working by setting up a new container entirely. I added the LinuxServer-style `PUID` and `PGID` environment variables, though I am not sure if they are required at this time (it seems like they are based on discussions in this thread, but they are not called out in the sample docker compose on the main github page).

I would also recommend against running any container in "privileged" mode (like some other users in this thread have done, as a workaround) due to the security risk it can create (though admittedly I do not entirely understand the full implications of running containers in that mode).
